File: google-nearby-messages-example/batterylevel/android/app/src/main/kotlin/com/example/batterylevel/NearbyMessages.java
package com.example.batterylevel;

import android.os.Bundle;
import android.util.Log;

import com.google.android.gms.nearby.Nearby;
import com.google.android.gms.nearby.messages.Message;
import com.google.android.gms.nearby.messages.MessageListener;

import io.flutter.embedding.android.FlutterActivity;

import static android.content.ContentValues.TAG;

public class NearbyMessages extends FlutterActivity {

    private Message mMessage;
    MessageListener mMessageListener;

    @Override
    public void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);

        MessageListener mMessageListener = new MessageListener() {
            @Override
            public void onFound(Message message) {
                Log.d(TAG, "Found message: " + new String(message.getContent()));
            }

            @Override
            public void onLost(Message message) {
                Log.d(TAG, "Lost sight of message: " + new String(message.getContent()));
            }
        };

        mMessage = new Message("Hello World".getBytes());
    }

    @Override
    public void onStart() {
        super.onStart();

        // ...
        
        Nearby.getMessagesClient(this).publish(mMessage);
        Nearby.getMessagesClient(this).subscribe(mMessageListener);
    }

    @Override
    public void onStop() {
        Nearby.getMessagesClient(this).unpublish(mMessage);
        Nearby.getMessagesClient(this).unsubscribe(mMessageListener);

        // ...

        super.onStop();
    }

}

File: google-nearby-messages-example/batterylevel/android/app/src/main/kotlin/com/example/batterylevel/NearbyMessages.kt
package com.example.batterylevel

import android.content.ContentValues
import android.os.Bundle
import android.util.Log
import com.google.android.gms.nearby.Nearby
import com.google.android.gms.nearby.messages.Message
import com.google.android.gms.nearby.messages.MessageListener
import io.flutter.embedding.android.FlutterActivity

class NearbyMessages : FlutterActivity() {
    private var mMessage: Message? = null
    var mMessageListener: MessageListener? = null
    
    public override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        val mMessageListener: MessageListener = object : MessageListener() {
            override fun onFound(message: Message) {
                Log.d(ContentValues.TAG, "Found message: " + String(message.content))
            }

            override fun onLost(message: Message) {
                Log.d(ContentValues.TAG, "Lost sight of message: " + String(message.content))
            }
        }
        mMessage = Message("Hello World".toByteArray())
    }

    public override fun onStart() {
        super.onStart()

        // ...
        Nearby.getMessagesClient(this).publish(mMessage!!)
        Nearby.getMessagesClient(this).subscribe(mMessageListener!!)
    }

    public override fun onStop() {
        Nearby.getMessagesClient(this).unpublish(mMessage!!)
        Nearby.getMessagesClient(this).unsubscribe(mMessageListener!!)

        // ...
        super.onStop()
    }
}